src/main/java/hlf/java/rest/client/service/impl/EventPublishServiceImpl.java

```diff
@@ -86,9 +91,17 @@ public boolean publishChaincodeEvents(
     boolean status = true;
 
     try {
-
+        String key = String.valueOf(payload.hashCode());
+        if (fabricProperties.getEvents().isStandardCCEventEnabled()) {
+            // Fetch the key information for chaincode events,
+            // but only if the feature is enabled.
+
+            // Parse the payload and use the key.
+            StandardCCEvent standardCCEvent = FabricEventParseUtil.parseString(payload, StandardCCEvent.class);
+            key = standardCCEvent.getKey();
+        }
       ProducerRecord<String, String> producerRecord =
-          new ProducerRecord<>(topicName, String.valueOf(payload.hashCode()), payload);
+          new ProducerRecord<>(kafkaProperties.getEventListener().getTopic(), key, payload);
```

Comment:
wouldn't this result in the Event being sent in new`payload` structure? 
I assume we might have to keep it as new `ProducerRecord<>(kafkaProperties.getEventListener().getTopic(), key, standardCCEvent.getEvent());` so that the event gets routed based on the Key yet the deserialisation logic in downstream system remains the same.

Reply:
Or we can re-assign payload as `payload=standardCCEvent.getEvent()` just after line#100